From 9bbd613c027fe0ceedf365463eae07964bd00b87 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 6 Jan 2021 15:18:03 +0100
Subject: [PATCH 1/1] MdeModulePkg: correct TPL level check CoreCreateEventEx

REF: https://bugzilla.tianocore.org/show_bug.cgi?id=3058

According to table Table 27. TPL Restrictions in UEFI spec 2.8 B:

Event Notification Levels must be > TPL_APPLICATION and <= TPL_HIGH_LEVEL.

Adjust the test TPL level test.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 MdeModulePkg/Core/Dxe/Event/Event.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/MdeModulePkg/Core/Dxe/Event/Event.c b/MdeModulePkg/Core/Dxe/Event/Event.c
index c83c572c8f84..e25856eaaa5f 100644
--- a/MdeModulePkg/Core/Dxe/Event/Event.c
+++ b/MdeModulePkg/Core/Dxe/Event/Event.c
@@ -339,9 +339,9 @@ CoreCreateEventEx (
   // If it's a notify type of event, check for invalid NotifyTpl
   //
   if ((Type & (EVT_NOTIFY_WAIT | EVT_NOTIFY_SIGNAL)) != 0) {
-    if (NotifyTpl != TPL_APPLICATION &&
-        NotifyTpl != TPL_CALLBACK &&
-        NotifyTpl != TPL_NOTIFY) {
+    if (NotifyTpl != TPL_CALLBACK &&
+        NotifyTpl != TPL_NOTIFY &&
+        NotifyTpl != TPL_HIGH_LEVEL) {
       return EFI_INVALID_PARAMETER;
     }
   }
-- 
2.29.2

